compiler/rustc_borrowck/src/type_check/liveness/trace.rs

```diff
@@ -555,16 +558,68 @@ impl<'tcx> LivenessContext<'_, '_, '_, 'tcx> {
             values::location_set_str(elements, live_at.iter()),
         );
 
-        let tcx = typeck.tcx();
-        tcx.for_each_free_region(&value, |live_region| {
-            let live_region_vid =
-                typeck.borrowck_context.universal_regions.to_region_vid(live_region);
-            typeck
-                .borrowck_context
-                .constraints
-                .liveness_constraints
-                .add_elements(live_region_vid, live_at);
-        });
+        struct MakeAllRegionsLive<'a, 'b, 'tcx> {
+            typeck: &'b mut TypeChecker<'a, 'tcx>,
+            live_at: &'b IntervalSet<PointIndex>,
+        }
+        impl<'tcx> MakeAllRegionsLive<'_, '_, 'tcx> {
+            fn make_alias_live(&mut self, t: Ty<'tcx>) -> ControlFlow<!> {
+                let ty::Alias(_kind, alias_ty) = t.kind() else {
+                    bug!();
+                };
+                let tcx = self.typeck.infcx.tcx;
+                let mut outlives_bounds = tcx
+                    .item_bounds(alias_ty.def_id)
+                    .iter_instantiated(tcx, alias_ty.args)
+                    .filter_map(|clause| {
+                        if let Some(outlives) = clause.as_type_outlives_clause()
+                        && outlives.skip_binder().0 == t
+                    {
+                        Some(outlives.skip_binder().1)
```

Comment:
What happens with `impl for<'a> 'a` ? I guess we probably don't accept that. Related question, does this take into account elaborated bounds? (In that case, you could have `impl for<'a> Trait<'a>` where `trait Trait<'a>: 'a`, for example.)

Reply:
> Related question, does this take into account elaborated bounds? (In that case, you could have impl for<'a> Trait<'a> where trait Trait<'a>: 'a, for example.)

It acts like a `for<'a> T: 'a` bound, so it'll be treated like `T: 'static` due to this code:

https://github.com/rust-lang/rust/blob/8e47113d71ed3922084571d70deba661f0a20001/compiler/rustc_infer/src/infer/outlives/test_type_match.rs#L51-L60

> What happens with `impl for<'a> 'a` ? I guess we probably don't accept that. 

We don't accept that in the grammar, but yeah, you can get it implied like above.